### ISSUE TITLE
Nats bootstrap failing on Windows

### ISSUE THREAD

**Issue by circlespainter** [YunaBraska/nats-streaming-server-embedded#15]
I get this exception:

```
Caused by: java.lang.IllegalArgumentException: Failed to initialise EnableNatsServer
	at berlin.yuna.natsserver.annotation.EnableNatsServerContextCustomizer.customizeContext(EnableNatsServerContextCustomizer.java:57) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at org.springframework.boot.test.context.SpringBootContextLoader$ContextCustomizerAdapter.initialize(SpringBootContextLoader.java:337) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:633) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:373) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:139) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:117) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	... 24 more
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at berlin.yuna.natsserver.util.SystemUtil.copyResourceFile(SystemUtil.java:73) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.logic.NatsServer.getNatsServerPath(NatsServer.java:201) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.logic.NatsServer.start(NatsServer.java:131) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.annotation.EnableNatsServerContextCustomizer.customizeContext(EnableNatsServerContextCustomizer.java:55) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at org.springframework.boot.test.context.SpringBootContextLoader$ContextCustomizerAdapter.initialize(SpringBootContextLoader.java:337) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:633) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:373) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:139) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:117) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	... 24 more
Caused by: java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Objects.java:203) ~[?:1.8.0_181]
	at java.nio.file.Files.copy(Files.java:2984) ~[?:1.8.0_181]
	at berlin.yuna.natsserver.util.SystemUtil.copyResourceFile(SystemUtil.java:70) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.logic.NatsServer.getNatsServerPath(NatsServer.java:201) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.logic.NatsServer.start(NatsServer.java:131) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at berlin.yuna.natsserver.annotation.EnableNatsServerContextCustomizer.customizeContext(EnableNatsServerContextCustomizer.java:55) ~[nats-streaming-server-embedded-0.2.1.jar:?]
	at org.springframework.boot.test.context.SpringBootContextLoader$ContextCustomizerAdapter.initialize(SpringBootContextLoader.java:337) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:633) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:373) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:139) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:117) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	... 24 more
```

I think the problem is in `NatsServer.getNatsServerPath`: if the path component separator is a regular forward-slash then `SystemUtil.class.getClassLoader().getResourceAsStream(relativeResource)` evaluates to an actual `InputStream` rather then `null` (at least in the debugger).

**Comment by YunaBraska** [comment]
Thanks, I will test that...

**Comment by YunaBraska** [comment]
Okay, i was setting up a windows machine and were running all the tests. you are right there is a path problem. i was assuming that java nit would fix the path problem on windows with the ":" but it's not. :(

**Comment by YunaBraska** [comment]
Fixed with #16
